pkg/transfer/s3uploader.go

```diff
@@ -60,6 +61,7 @@ func NewS3(cfg *S3Conf) (trans *S3, err error) {
 	//setup the official uploader
 	trans.upl = s3manager.NewUploaderWithClient(s3api)
 	trans.dwn = s3manager.NewDownloaderWithClient(s3api)
+	trans.del = s3api
```

Comment:
Lets call this "trans.api", the trans.upl and trans.dwn are exceptions here as they happen to be API's for uploading and downloading